src/platform_impl/web/web_sys/canvas/pointer_handler.rs

```diff
@@ -105,11 +115,113 @@ impl PointerHandler {
         ));
     }
 
+    pub fn on_touch_move<F>(&mut self, canvas_common: &super::Common, mut handler: F)
+    where
+        F: 'static + FnMut(i32, PhysicalPosition<f64>, Force),
+    {
+        let canvas = canvas_common.raw.clone();
+        self.on_touch_move = Some(canvas_common.add_event(
+            "pointermove",
+            move |event: PointerEvent| {
+                if event.pointer_type() != "touch" {
+                    return;
+                }
+
+                handler(
+                    event.pointer_id(),
+                    touch_physical_position(&event, &canvas),
+                    Force::Normalized(event.pressure() as f64),
+                );
+            },
+        ));
+    }
+
+    pub fn on_touch_down<F>(&mut self, canvas_common: &super::Common, mut handler: F)
+    where
+        F: 'static + FnMut(i32, PhysicalPosition<f64>, Force),
+    {
+        let canvas = canvas_common.raw.clone();
+        self.on_touch_down = Some(canvas_common.add_event(
+            "pointerdown",
+            move |event: PointerEvent| {
+                if event.pointer_type() != "touch" {
+                    return;
+                }
+
+                handler(
+                    event.pointer_id(),
+                    touch_physical_position(&event, &canvas),
+                    Force::Normalized(event.pressure() as f64),
+                );
+            },
+        ));
+    }
+
+    pub fn on_touch_up<F>(&mut self, canvas_common: &super::Common, mut handler: F)
+    where
+        F: 'static + FnMut(i32, PhysicalPosition<f64>, Force),
+    {
+        let canvas = canvas_common.raw.clone();
+        self.on_touch_up = Some(canvas_common.add_event(
+            "pointerup",
+            move |event: PointerEvent| {
+                if event.pointer_type() != "touch" {
+                    return;
+                }
+
+                handler(
+                    event.pointer_id(),
+                    touch_physical_position(&event, &canvas),
+                    Force::Normalized(event.pressure() as f64),
+                );
+            },
+        ));
+    }
+
+    pub fn on_touch_cancel<F>(&mut self, canvas_common: &super::Common, mut handler: F)
+    where
+        F: 'static + FnMut(i32, PhysicalPosition<f64>, Force),
+    {
+        let canvas = canvas_common.raw.clone();
+        self.on_touch_cancel = Some(canvas_common.add_event(
+            "pointercancel",
+            move |event: PointerEvent| {
+                if event.pointer_type() != "touch" {
+                    return;
+                }
+
+                handler(
+                    event.pointer_id(),
+                    touch_physical_position(&event, &canvas),
+                    Force::Normalized(event.pressure() as f64),
+                );
+            },
```

Comment:
This closure is duplicated between all the touch events, so it might be a good idea to extract it out into another method.

Reply:
I resolved this